rmgweb/database/views.py

```diff
@@ -3484,6 +3493,408 @@ def solvationSoluteSearch(request):
+    solute and temperature. Reference solvent is
+    """
+    from rmgweb.database.forms import SolubilitySearchForm
+    table_rows = 10
+    solubility_form_set_factory = formset_factory(SolubilitySearchForm, extra=table_rows)
+    solubility_form_set = solubility_form_set_factory()
+    smiles_error_msg, temp_error_msg, ref_solubility_error_msg, hsub_error_msg, cp_error_msg, required_input_error_msg,\
+    optional_input_error_msg, empty_error_msg = False, False, False, False, False, False, False, False
+
+    if request.method == 'POST':
+        posted = solubility_form_set_factory(request.POST)
+        initial = request.POST.copy()
+
+        if posted.is_valid():
+            # initialize the input dictionary to pass to the result page
+            str_input_dict = {}
+            str_input_name_list = ['solvent_str', 'solute_str', 'temp_str', 'ref_solvent_str', 'ref_solubility_str',
+                                   'ref_temp_str', 'hsub298_str', 'cp_gas_298_str', 'cp_solid_298_str']
+            for str_input_name in str_input_name_list:
+                str_input_dict[str_input_name] = None
+            # get the input from the posted formset
+            for form in posted:
+                cd = form.cleaned_data
+                solvent, solute, temp, ref_solvent, ref_solubility, ref_temp, hsub298, cp_gas_298, cp_solid_298 = \
+                    cd.get('solvent'), cd.get('solute'), cd.get('temp'), cd.get('ref_solvent'), cd.get('ref_solubility'), \
+                    cd.get('ref_temp'), cd.get('hsub298'), cd.get('cp_gas_298'), cd.get('cp_solid_298')
+                input_value_list = [solvent, solute, temp, ref_solvent, ref_solubility, ref_temp, hsub298, cp_gas_298,
+                                    cp_solid_298]
+                # skip if all values are None:
+                if not any(input_value_list):
+                    continue
+                # check for invalid input and add appropriate error messages
+                required_input_list = [solvent, solute, temp]
+                optional_input_set_list = [ref_solvent, ref_solubility, ref_temp]
+                if not all(required_input_list):
+                    required_input_error_msg = parse_input_error_msg(
+                        required_input_error_msg, 'Solvent SMILES, Solute SMILES, and Temp are required inputs')
+                if any(optional_input_set_list) and not all (optional_input_set_list):
+                    optional_input_error_msg = parse_input_error_msg(
+                        optional_input_error_msg, 'All three reference values (ref. solvent SMILES, ref. solubility, '
+                                                  'ref. temp) must be provided together')
+                if '_' in str(solvent) or '_' in str(solute) or '_' in str(ref_solvent):
+                    smiles_error_msg = parse_input_error_msg(smiles_error_msg, 'An underscore (_) cannot be in SMILES')
+                if temp is not None:
```

Comment:
Thanks for suggestion! However, if I use this approach, the error message variables themselves are not getting updated and it is the dictionary "message' value that is being updated. This will make me assign each error message variable later like this:
``` 
temp_error_msg = input_checks_dict['temp']['message']
ref_solubility_error_msg = input_checks_dict['ref_solubility']['message']
...
```

Another problem is that `temp` and `ref_temp` share the same error message variable like `temp_error_msg`, but the proposed method won't be able to handle this properly as each error message is stored as a separate variable inside the dictionary. 

So I would keep this part of the code as it is unless you have any other suggestion.

Reply:
OK! Sounds good.